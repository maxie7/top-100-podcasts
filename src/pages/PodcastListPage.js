import { useState, useEffect } from 'react';
import Header from '../components/Header';
import PodcastListItems from '../components/PodcastListItems';
import PodcastListFilter from '../components/PodcastListFilter';
import { getInitialValues } from '../utils/getInitialValues';
import { getPodcasts } from '../core/application/getPodcastsService';

const PodcastListPage = () => {
  const [error, setError] = useState(null);
  const [isLoaded, setIsLoaded] = useState(false);
  const [items, setItems] = useState([]);
  const [query, setQuery] = useState('');
  const [podcasts, setPodcasts] = useState(getInitialValues('podcastsList'));

  useEffect(() => {
    const now = new Date().getTime();
    const getPodcastsData = async () => {
      const podcastsData = await getPodcasts();
      if (podcastsData.error) {
        setIsLoaded(true);
        setError(podcastsData.error);
      } else {
        setPodcasts(podcastsData);
        setItems(podcastsData.data.feed.entry);
        setIsLoaded(true);
      };
    };

    if (!podcasts || now > podcasts.expiry) {
      getPodcastsData();
    } else {
      setIsLoaded(true);
      setItems(podcasts.value);
    }
  }, []);

  if (error) {
    return console.log('REQ ERROR:', error);
  } else {
    return (
      <>
        <Header isLoaded={isLoaded} />
        <div className="primary">
          <PodcastListFilter query={query} setQuery={setQuery} podcastsLength={items.length} />
          <PodcastListItems items={items} query={query} />
        </div>
      </>
    );
  }
};

export default PodcastListPage;
