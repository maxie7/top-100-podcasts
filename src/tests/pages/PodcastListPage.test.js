import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import PodcastListPage from '../../pages/PodcastListPage';

describe('PodcastListPage component', () => {
  it('should render component Podcast List Page', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<PodcastListPage />);

    expect(renderer.render(<PodcastListPage />)).toMatchSnapshot();
  });
});
