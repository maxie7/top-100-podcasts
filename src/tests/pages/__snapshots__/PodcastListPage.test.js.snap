// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastListPage component should render component Podcast List Page 1`] = `
<React.Fragment>
  <Header
    isLoaded={false}
  />
  <div
    className="primary"
  >
    <PodcastListFilter
      query=""
      setQuery={[Function]}
    />
    <PodcastListItems
      items={Array []}
      query=""
    />
  </div>
</React.Fragment>
`;
